password-cli/README.md

```diff
@@ -1,22 +1,99 @@
-# password-cli
+# `password-cli`
 
 [![Build Status](https://github.com/cdepillabout/password/workflows/password/badge.svg)](http://github.com/cdepillabout/password)
 [![Hackage](https://img.shields.io/hackage/v/password-cli.svg)](https://hackage.haskell.org/package/password-cli)
 [![Stackage LTS](http://stackage.org/package/password-cli/badge/lts)](http://stackage.org/lts/package/password-cli)
 [![Stackage Nightly](http://stackage.org/package/password-cli/badge/nightly)](http://stackage.org/nightly/package/password-cli)
 [![BSD3 license](https://img.shields.io/badge/license-BSD3-blue.svg)](./LICENSE)
 
-This package provides a simple CLI for [password](https://hackage.haskell.org/package/password) package.
+This package provides a simple CLI for the [`password`](https://hackage.haskell.org/package/password) package.
+As such it supports all the algorithms that the [`password`](https://hackage.haskell.org/package/password)
+package supports, which at the time of writing are `Argon2`, `brypt`, `PBKDF2` and `scrypt`.
 
-Pipeline usage:
+At the moment, the default settings are used for each algorithm, but this will probably become configurable in
+a later version of the CLI.
 
+## Example usage
+
+The following sections give examples of how the CLI can be used.
+
+### Hashing a password interactively
+
+Hashing a password interactively is as easy as
+```
+user@computer $ password-cli hash bcrypt
+Enter password:
+```
+where the input is then hidden and the hash is printed to the screen, resulting in
+```
+user@computer $ password-cli hash bcrypt
+Enter password:
+$2b$10$JuNbIWqVQD2EldT481zEEuaVKROrYhsHXLjM/Tx3e7ahJQxVw7N4y
+user@computer $
+```
+
+### Hashing a password with pipes
+
+When piping in the password from a file or other program:
+```
+user@computer $ cat password.txt | password-cli hash pbkdf2
+Enter password:
+sha512:25000:8ZJ1T55Y0sPRwltXNe/2fA==:aA0BT1WlTg+t2pSr8E6+l2zJW88rmUiDlKeohSOnzS0nLOumDSyK0FfsiNJBvWvWVkB2r6IMxRqelk4LZR33ow==user@computer $
 ```
```

Comment:
````suggestion
```console
$ cat password.txt | password-cli hash pbkdf2
Enter password:
sha512:25000:8ZJ1T55Y0sPRwltXNe/2fA==:aA0BT1WlTg+t2pSr8E6+l2zJW88rmUiDlKeohSOnzS0nLOumDSyK0FfsiNJBvWvWVkB2r6IMxRqelk4LZR33ow==
```
````

I'd just remove the `user@computer $` at the end of the line here, since you explain it in the next paragraph.  

Although I see why you originally wrote it like this.  It may be easier to understand as you have it in this current PR.

Reply:
Yeah, but without the `user@computer` it's difficult to convey it obviously.
I removed it and hope it's obvious from the text and any testing people would do with the CLI.